drink/src/mock.rs

```diff
@@ -0,0 +1,36 @@
+mod contract;
+mod error;
+mod mocking_api;
+
+use std::collections::BTreeMap;
+
+pub use contract::{mock_message, ContractMock, MessageMock, Selector};
+use error::MockingError;
+pub use mocking_api::MockingApi;
+
+/// Untyped result of a mocked call.
+pub type MockedCallResult = Result<Vec<u8>, MockingError>;
+
+/// A registry of mocked contracts.
+pub(crate) struct MockRegistry<AccountId: Ord> {
+    mocked_contracts: BTreeMap<AccountId, ContractMock>,
+}
+
+impl<AccountId: Ord> MockRegistry<AccountId> {
+    /// Creates a new registry.
+    pub fn new() -> Self {
+        Self {
+            mocked_contracts: BTreeMap::new(),
+        }
+    }
+
+    /// Registers `mock` for `address`.
+    pub fn register(&mut self, address: AccountId, mock: ContractMock) {
+        self.mocked_contracts.insert(address, mock);
```

Comment:
[`BTreeMap::insert`](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.insert) returns `Option<V>` - previous element if any. Can we return it here as well? Just in case it turns out in the future that someone needs it. I don't see any reason why we should not.

Reply:
:white_check_mark: